src/potions.ts

```diff
@@ -417,3 +431,132 @@ export function bathroomFinance(embezzlers: number): void {
+    const barfTurns = Math.max(0, estimatedTurns() - yachtzees - embezzlers);
+
+    const potionAmountsToConsider: number[] = [];
+    const considerSoftcap = [0, this.softcap];
+    const considerEmbezzlers = embezzlers > 0 ? [0, embezzlers] : [0];
+    for (const fn of [Math.floor, Math.ceil]) {
+      for (const sc of considerSoftcap) {
+        for (const em of considerEmbezzlers) {
+          const considerBarfTurns = em === embezzlers && barfTurns > 0 ? [0, barfTurns] : [0];
+          for (const bt of considerBarfTurns) {
+            const potionAmount = fn((yachtzees + em + bt + sc) / this.duration);
+            if (!potionAmountsToConsider.includes(potionAmount)) {
+              potionAmountsToConsider.push(potionAmount);
+            }
+          }
+        }
+      }
+    }
+
+    const profitsFromPotions = potionAmountsToConsider.map((quantity) => ({
+      quantity,
+      value: this.valueNPotions(quantity, yachtzees, embezzlers, barfTurns),
+    }));
+    const bestOption = profitsFromPotions.reduce((a, b) => (a.value > b.value ? a : b));
+
+    if (bestOption.value > 0) {
+      print(
+        `Expected to profit ${bestOption.value.toFixed(2)} from ${bestOption.quantity} ${
+          this.potion.plural
+        }`,
+        "blue"
+      );
+      const potionsToUse =
+        bestOption.quantity - Math.floor(haveEffect(this.effect) / this.duration);
+      return Math.max(potionsToUse, 0);
+    }
+    return 0;
+  }
+
+  valueNPotions(n: number, yachtzees: number, embezzlers: number, barfTurns: number): number {
+    const yachtzeeValue = 2000;
+    const embezzlerValue = baseMeat + 750;
+    const barfValue = (baseMeat * turnsToNC) / 30;
+
+    const totalCosts = retrievePrice(this.potion, n);
+    const totalDuration = n * this.duration;
+    let cappedDuration = Math.max(0, totalDuration - this.softcap + 1);
+    let decayDuration = Math.min(totalDuration, this.softcap - 1);
+    let totalValue = 0;
+    const turnTypes = [
+      [yachtzees, yachtzeeValue],
+      [embezzlers, embezzlerValue],
+      [barfTurns, barfValue],
+    ];
+
+    for (const [turns, value] of turnTypes) {
+      const cappedTurns = Math.min(cappedDuration, turns);
+      const decayTurns = Math.min(decayDuration, turns - cappedTurns);
+      totalValue +=
+        (value *
+          (cappedTurns * this.cappedMeatBonus +
+            triangleNumber(decayDuration, decayDuration - decayTurns) * this.meatBonusPerTurn)) /
+        100;
+      cappedDuration -= cappedTurns;
+      decayDuration -= decayTurns;
+      if (decayDuration === 0) break;
+    }
+
+    return totalValue - totalCosts;
+  }
+}
+
+export function variableMeatPotionsSetup(yachtzees: number, embezzlers: number): void {
+  const potions = [
+    new VariableMeatPotion($item`love song of sugary cuteness`, 20, 2),
+    new VariableMeatPotion($item`pulled yellow taffy`, 50, 2),
+    // new VariableMeatPotions($item`porcelain candy dish`, 500, 1),
```

Comment:
Is there a reason this is commented out?

Reply:
I wasn't super comfortable playing with potentially millions of meat's worth of a buff, especially when we need at least 500 to get the ball rolling (and I have reservations about estimating prices of bulk purchases)